Mathlib/NumberTheory/ModularForms/EisensteinSeries/Basic.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import Mathlib.NumberTheory.ModularForms.SlashInvariantForms
+import Mathlib.NumberTheory.ModularForms.CongruenceSubgroups
+import Mathlib.Analysis.Normed.Field.InfiniteSum
+import Mathlib.LinearAlgebra.Matrix.SpecialLinearGroup
+
+/-!
+# Eisenstein Series
+
+## Main definitions
+
+* We define Eisenstein series of level `Γ(N)` for any `N : ℕ` and weight `k : ℤ` as the infinite sum
+  `∑' v : (Fin 2 → ℤ), (1 / (v 0 * z + v 1) ^ k)`, where `z : ℍ` and `v` ranges over all pairs of
+  coprime integers congruent to a fixed pair `(a, b)` modulo `N`.
+* We show that they define a slash invariant form of level `Γ(N)` and weight `k`.
+
+## References
+ See Diamond and Shurman, A First Course in Modular Forms.
+
+-/
+
+
+noncomputable section
+
+open ModularForm UpperHalfPlane Complex Matrix Classical Nat
+
+local notation "SL(" n ", " R ")" => Matrix.SpecialLinearGroup (Fin n) R
```

Comment:
Isn't this a scoped notation or something? It seems pretty useful.

Reply:
Sorry, I'm not sure about this.  Can you explain the difference between the local and scoped? I've never used scoped notation (I don't think).

Reply:
The notation already exists in mathlib but you need to do "open scoped MatrixGroups" to activate it. (This is a bit like "open_locale" in lean3.)

Reply:
My point was "this should be a scoped notation", even better if it already is.